src/metadataGeneration/tsoa.ts

```diff
@@ -68,7 +68,7 @@ export namespace Tsoa {
     validators: Validators;
   }
 
-  export type TypeStringLiteral = 'string' | 'boolean' | 'double' | 'float' | 'integer' | 'long' | 'enum' | 'array' | 'datetime' | 'date' | 'binary' | 'buffer' | 'byte' | 'void' | 'object' | 'any' | 'refEnum' | 'refObject';
+  export type TypeStringLiteral = 'string' | 'boolean' | 'double' | 'float' | 'integer' | 'long' | 'enum' | 'array' | 'datetime' | 'date' | 'binary' | 'buffer' | 'byte' | 'void' | 'object' | 'any' | 'refEnum' | 'refObject' | 'objectLiteral';
```

Comment:
What would you think about renaming it to be “nestedObject”? I ask this because we already have the “Object” type and the “refObject” so it makes for some confusing code. I realize that the type your describing is most accurately described as an objectLiteral, but I think a different name might lead to more clear code for future contributors.

Reply:
Another option is “InLineObject”

Reply:
I understand your concern for new contributors, I was siding more with the expectation of someone comfortable with the internal TS terminology with my initial proposal.
Maybe as a 
Is `nestedObjectTypeLiteral` something you would agree on as a compromise between the two?

Reply:
Yes, `nestedObjectTypeLiteral` is a wonderful compromise. :)